plugin/src/main/kotlin/com/ritense/plugin/service/PluginService.kt

```diff
@@ -85,9 +88,19 @@ class PluginService(
         val pluginDefinition = pluginDefinitionRepository.getById(pluginDefinitionKey)
         validateProperties(properties, pluginDefinition)
 
-        return pluginConfigurationRepository.save(
+        val pluginConfiguration = pluginConfigurationRepository.save(
             PluginConfiguration(PluginConfigurationId.newId(), title, properties, pluginDefinition)
         )
+
+        try {
+            pluginConfiguration.runAllPluginEvents(PluginEvent.EventType.CREATE)
+
+        } catch (e: Exception) {
+            pluginConfigurationRepository.deleteById(pluginConfiguration.id)
```

Comment:
The method 'createPluginConfiguration' can be `@Transactional`. 

That way the pluginConfiguration won't have to be deleted here. Because the exception will trigger a rollback which will rollback the `.save()`

Reply:
Yes, we tried this and discussed this with Marijn when fixing existing Integratin and Unit Tests.
Let's discuss this.